src/main.cpp

```diff
@@ -33,63 +37,136 @@ constexpr auto VERSION = _VERSION;
 #else
 constexpr auto VERSION = "[Unkown version]";
 #endif
+class InputParser {
+ public:
+  InputParser(int& argc, char** argv) {
+    for (int i = 1; i < argc; ++i) this->tokens.push_back(std::string(argv[i]));
+  }
 
-void printWelcomeMessage() {
-  std::cout << "Touchégg " << VERSION << "." << std::endl;
-  std::cout << "Usage: touchegg [--daemon [start_threshold finish_threshold]] "
-               "[--client]"
-            << std::endl
-            << std::endl;
-
-  std::cout << "Multi-touch gesture recognizer." << std::endl;
-  std::cout << "Touchégg is an app that runs in the background and transform "
-               "the gestures you make on your touchpad into visible actions in "
-               "your desktop."
-            << std::endl;
-  std::cout << "For more information please visit:" << std::endl;
-  std::cout << "https://github.com/JoseExposito/touchegg" << std::endl
-            << std::endl;
-
-  std::cout << "Option\t\tMeaning" << std::endl;
-  std::cout << " --daemon\tRun Touchégg in daemon mode. This mode starts a "
-               "service that gathers gestures but executes no actions"
-            << std::endl;
-  std::cout << " --client\tConnect to an existing Touchégg daemon and "
-               "execute actions in your desktop"
-            << std::endl;
-  std::cout << "Without arguments Touchégg starts in client mode" << std::endl
-            << std::endl;
+  const std::string getCmdOption(const std::string& option) const {
+    std::vector<std::string>::const_iterator itr;
+    itr = std::find(this->tokens.begin(), this->tokens.end(), option);
+    if (itr != this->tokens.end() && ++itr != this->tokens.end()) {
+      return *itr;
+    }
+    static const std::string empty_string("");
+    return empty_string;
+  }
+
+  // special case used for daemon thresholds
+  void getCmdOption2d(const std::string& option, double& d1, double& d2) const {
+    std::vector<std::string>::const_iterator itr;
+    itr = std::find(this->tokens.begin(), this->tokens.end(), option);
+    if (itr != this->tokens.end()) {
+      if (++itr != this->tokens.end()) {
+        d1 = std::stod(*itr);
+        if (++itr != this->tokens.end()) {
+          d2 = std::stod(*itr);
+        }
+      }
+    }
+  }
+
+  bool cmdOptionExists(const std::string& option) const {
+    return std::find(this->tokens.begin(), this->tokens.end(), option) !=
+           this->tokens.end();
+  }
+
+ private:
+  std::vector<std::string> tokens;
+};
+
+// Parse the command line arguments
+void parseArgs(int argc, char** argv, bool& daemonMode, bool& clientMode,
```

Comment:
This function could be moved to the `ArgsParser` class, or even be part of the constructor.

Reply:
Hmm, it could be part of the class but that would couple the class to this program specifically. I don't think that's a great idea but if you insist I'll do it.